internal/peer/node/throttle_grpc.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package node
+
+import (
+	"context"
+
+	"github.com/hyperledger/fabric/common/semaphore"
+	"github.com/hyperledger/fabric/core/peer"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+func initGrpcSemaphores(config *peer.Config) map[string]semaphore.Semaphore {
+	semaphores := make(map[string]semaphore.Semaphore)
+	endorserConcurrency := config.LimitsConcurrencyEndorserService
+	deliverConcurrency := config.LimitsConcurrencyDeliverService
+
+	if endorserConcurrency != 0 {
+		semaphores["/protos.Endorser/ProcessProposal"] = semaphore.New(endorserConcurrency)
+	}
+
+	if deliverConcurrency != 0 {
+		deliverSemaphore := semaphore.New(deliverConcurrency)
+		semaphores["/protos.Deliver/Deliver"] = deliverSemaphore
+		semaphores["/protos.Deliver/DeliverFiltered"] = deliverSemaphore
+		semaphores["/protos.Deliver/DeliverWithPrivateData"] = deliverSemaphore
+	}
+
+	return semaphores
+}
+
+func unaryThrottleInterceptor(semaphores map[string]semaphore.Semaphore) grpc.UnaryServerInterceptor {
```

Comment:
Nit.
```suggestion
func unaryGrpcLimiter(semaphores map[string]semaphore.Semaphore) grpc.UnaryServerInterceptor {
```
? Throttle and limit are used interchangeably. It is better to use a single term consistency -- `Limit` for both file name and variable names (for better readability). Difference between throttling and rate-limiting -- https://helpx.adobe.com/coldfusion/api-manager/throttling-and-rate-limiting.html

Reply:
Renamed both file name and method names.